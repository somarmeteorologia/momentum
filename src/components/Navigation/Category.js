import React, { useContext } from 'react'
import styled, { ThemeContext } from 'styled-components'
import { theme } from 'styled-tools'

const Container = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: 15px;
  padding: 15px 20px;
  background-color: ${theme('navigation.bg.secondary')};
`

const Title = ({ children }) => {
  const { navigation } = useContext(ThemeContext)
  const color = navigation.text.primary

  return (
    <Container>{children({ details: color, description: color })}</Container>
  )
}

export default {
  id: 'category',
  Title
}
