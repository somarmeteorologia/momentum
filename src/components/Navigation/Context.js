import React, { createContext, useState, useEffect } from 'react'

const Context = createContext()

function Provider({ value, children }) {
  const { defaultStructure, defaultInterables, onInterable } = value
  const [structure, setStructure] = useState(defaultStructure)
  const [interables, setInterables] = useState(defaultInterables)

  useEffect(() => {
    onInterable({ interables, setInterables })
  }, [interables])

  return (
    <Context.Provider
      value={{
        structure,
        setStructure,
        interables,
        setInterables
      }}
    >
      {children}
    </Context.Provider>
  )
}

const Consumer = Context.Consumer

export { Consumer, Context, Provider }
