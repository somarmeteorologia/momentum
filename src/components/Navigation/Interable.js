import styled from 'styled-components'

export default styled.div.attrs({ 'data-interable': 'interable' })`
  display: inline-block;
  padding: 0 5px;
`
