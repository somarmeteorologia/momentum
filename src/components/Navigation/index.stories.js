import React from 'react'
import styled from 'styled-components'
import { storiesOf } from '@storybook/react'

import { GROUPS } from '@helpers/hierarchySeparators'

import { Reset } from '@components/Reset'

import Testeable from './Testeable'

const Container = styled.div`
  width: 100%;
  height: 100vh;
`

const Default = ({ toOpen }) => {
  return (
    <Container>
      <Testeable toOpen={toOpen} />
    </Container>
  )
}

const AtSpecific = () => {
  const toOpen = () => 'newsletter'

  return <Default toOpen={toOpen} />
}

storiesOf(`${GROUPS.COMPONENTS}|Navigation`, module)
  .addDecorator(story => (
    <>
      <Reset />
      {story()}
    </>
  ))
  .add('Default', () => <Default />)
  .add('At specific', () => <AtSpecific />)
