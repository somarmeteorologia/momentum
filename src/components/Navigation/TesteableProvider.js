import React from 'react'

import { Navigation } from '@components/Navigation'
import { Switch } from '@components/Switch'
import { Text } from '@components/Text'
import { Icon } from '@components/Icon'

const {
  Type,
  Interable,
  Separator,
  Title,
  Control,
  useInterable,
  Provider
} = Navigation

const defaultInterables = [
  {
    id: 'cars',
    interables: [
      {
        id: 'cars',
        value: true
      }
    ]
  },
  {
    id: 'volkswagen',
    interables: [
      {
        id: 'switch',
        value: true
      }
    ]
  },
  {
    id: 'ford',
    interables: [
      {
        id: 'switch',
        value: true
      }
    ]
  },
  {
    id: 'gol',
    interables: [
      {
        id: '1.6',
        value: true
      },
      {
        id: '1.0',
        value: false
      }
    ]
  },
  {
    id: 'knife',
    interables: [
      {
        id: 'knife',
        value: false
      }
    ]
  },
  {
    id: 'nothing',
    interables: [
      {
        id: 'isVisible',
        value: true
      }
    ]
  }
]

const defaultStructure = [
  {
    id: 'monitoring',
    type: Type.Category,
    title: ({ details, description }) => (
      <>
        <Icon name="home" right={10} width={20} height={20} color={details} />
        <Text
          weight={Text.weight.bold}
          size={Text.size.fourteen}
          color={description}
        >
          Monitoramento
        </Text>
      </>
    ),
    children: [
      {
        id: 'cars',
        parent: 'monitoring',
        onPrevent: () => console.log('prevented'),
        title: ({ getter, setter }) => (
          <Interable>
            <>
              <Switch
                id="cars"
                label="Cars"
                labelAlign="right"
                active={getter('cars')}
                onChange={() => setter('cars', !getter('cars'))}
              />
            </>
          </Interable>
        ),
        type: Type.Group,
        children: [
          {
            id: 'volkswagen',
            type: Type.Group,
            parent: 'cars',
            title: ({ getter, setter }) => (
              <Interable>
                <Switch
                  id="switch"
                  label="Volkswagen"
                  labelAlign="right"
                  active={getter('switch')}
                  onChange={() => setter('switch', !getter('switch'))}
                />
              </Interable>
            ),
            children: [
              {
                id: 'gol',
                parent: 'volkswagen',
                title: ({ getter, setter }) => (
                  <Control>
                    <>
                      <Switch
                        id="1.6"
                        label="1.6"
                        labelAlign="right"
                        active={getter('1.6')}
                        onChange={() => setter('1.6', !getter('1.6'))}
                      />

                      <Switch
                        id="1.0"
                        label="1.0"
                        labelAlign="right"
                        active={getter('1.0')}
                        onChange={() => setter('1.0', !getter('1.0'))}
                      />
                    </>
                  </Control>
                ),
                type: Type.Item,
                children: []
              }
            ]
          },
          {
            id: 'ford',
            parent: 'cars',
            title: ({ getter, setter }) => (
              <Interable>
                <Switch
                  id="switch"
                  label="Ford"
                  labelAlign="right"
                  active={getter('switch')}
                  onChange={() => setter('switch', !getter('switch'))}
                />
              </Interable>
            ),
            type: Type.Group,
            children: [
              {
                id: 'nothing',
                type: Type.Item,
                parent: 'ford',
                title: ({ getter }) => (
                  <div>{getter('isVisible') && 'Unavailable'}</div>
                ),
                children: []
              }
            ]
          }
        ]
      },
      {
        id: 'firstSeparator',
        parent: 'monitoring',
        title: () => <Separator />,
        type: Type.Item,
        children: []
      },
      {
        id: 'newsletter',
        parent: 'monitoring',
        title: ({ details }) => (
          <Title>
            <Icon
              name="newsletter"
              right={10}
              width={20}
              height={20}
              color={details}
            />
            <Text size={Text.size.fourteen}>Boletins</Text>
          </Title>
        ),
        type: Type.Group,
        children: [
          {
            id: 'content',
            parent: 'newsletter',
            type: Type.Item,
            title: () => <Text>Content from newsletter</Text>,
            children: []
          }
        ]
      }
    ]
  }
]

export default function TesteableProvider({ children }) {
  const onInterable = ({ interables, setIterables }) => {
    const withInterable = useInterable({ interables, setIterables })

    const [setter, getter] = withInterable('nothing')
  }

  return (
    <Provider
      defaultStructure={defaultStructure}
      defaultInterables={defaultInterables}
      onInterable={onInterable}
    >
      {children}
    </Provider>
  )
}
