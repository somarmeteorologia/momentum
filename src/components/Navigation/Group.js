import React, { useContext, useState } from 'react'
import PropTypes from 'prop-types'
import styled, { ThemeContext } from 'styled-components'
import { prop } from 'styled-tools'
import { Icon } from '@components/Icon'

import { Context } from './Context'
import { setOpenById, useInterable } from './utils'

const Container = styled.div`
  width: calc(100% - 10px);
  height: 50px;
  border-radius: ${prop('theme.border.radius.four')};
  line-height: 50px;
  cursor: pointer;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  margin-left: 5px;
  margin-bottom: 10px;

  &:hover {
    background-color: ${prop('theme.navigation.bg.hover')};
  }

  &:hover path,
  &:hover span {
    color: ${prop('theme.navigation.text.hover')};
  }
`

const Title = ({ id, children, onPrevent, prevent, onClick }) => {
  const [hover, setHover] = useState(false)
  const { navigation } = useContext(ThemeContext)

  const details = hover ? navigation.text.hover : navigation.bg.tertiary

  const { structure, setStructure, interables, setInterables } = useContext(
    Context
  )

  const [setter, getter] = useInterable({ interables, setInterables })(id)

  const toOpen = ({ target, currentTarget }) => {
    const interable = [].find.call(currentTarget.childNodes, node =>
      node.hasAttribute('data-interable')
    )

    if (interable && interable.contains(target)) {
      return
    }

    onClick && onClick()

    if (prevent({ getter })) {
      return onPrevent && onPrevent()
    }

    setStructure(setOpenById(id)(structure))
  }

  return (
    <Container
      data-testid={`open-${id}`}
      onClick={toOpen}
      onMouseEnter={() => setHover(true)}
      onMouseLeave={() => setHover(false)}
    >
      {children({
        getter,
        setter,
        details
      })}

      <Icon
        name="right"
        color={hover ? navigation.text.hover : navigation.bg.primary}
        width={12}
        height={12}
      />
    </Container>
  )
}

Title.defaultProps = {
  prevent: () => {},
  onClick: () => {}
}

Title.propTypes = {
  prevent: PropTypes.func,
  onPrevent: PropTypes.func,
  onClick: PropTypes.func,
  id: PropTypes.string.isRequired
}

export default {
  id: 'group',
  Title
}
