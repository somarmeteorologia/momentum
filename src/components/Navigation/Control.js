import React, { useState, useContext } from 'react'
import PropTypes from 'prop-types'
import styled, { ThemeContext, css } from 'styled-components'
import { ifProp } from 'styled-tools'

import { Text } from '@components/Text'
import { Icon } from '@components/Icon'

import Separator from './Separator'
import Interable from './Interable'

const Container = styled.div`
  display: flex;
  flex-direction: column;
`

const Content = styled.div`
  ${ifProp(
    { toggled: true },
    css`
      padding: 10px 15px;
    `
  )}
`

const Header = styled(Text)`
  cursor: pointer;
  display: flex;
  justify-content: flex-start;
  align-items: center;
`

export default function Control({ open, children }) {
  const { navigation } = useContext(ThemeContext)
  const [toggled, setToggled] = useState(open)

  const toToggle = ({ target, currentTarget }) => {
    const interable = [].find.call(currentTarget.childNodes, node =>
      node.hasAttribute('data-interable')
    )

    if (interable && interable.contains(target)) {
      return
    }

    setToggled(!toggled)
  }

  return (
    <Container onClick={toToggle}>
      <Header
        size={Text.size.twelve}
        color={navigation.text.primary}
        weight={Text.weight.bold}
        bottom={5}
      >
        <Icon
          name={toggled ? 'less' : 'more'}
          width={10}
          height={10}
          right={5}
        />
        Controle
      </Header>
      <Interable>
        <Content toggled={toggled}>{toggled && children}</Content>
      </Interable>
      <Separator />
    </Container>
  )
}

Control.defaultProps = {
  open: true
}

Control.propTypes = {
  open: PropTypes.bool.isRequired
}
