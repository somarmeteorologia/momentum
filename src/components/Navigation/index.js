import React, { useContext, useState, useRef } from 'react'
import styled, { ThemeContext } from 'styled-components'
import PropTypes from 'prop-types'
import { prop, ifProp } from 'styled-tools'
import posed from 'react-pose'

import { Icon } from '@components/Icon'

import Interable from './Interable'
import Control from './Control'
import Separator from './Separator'

import { Provider } from './Context'
import Switcher from './Switcher'
import Type from './Type'
import { useInterable } from './utils'

const Container = posed(styled.div`
  background-color: ${prop('theme.navigation.bg.primary')};
  height: ${ifProp('height', prop('height'), '100%')};
  position: relative;
`)({
  visible: {
    width: 270
  },
  collapsed: {
    width: 20
  }
})

const Title = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
`

const Toggleable = styled.div`
  width: 23px;
  height: 23px;
  cursor: pointer;
  border: 1px solid ${prop('theme.navigation.toggle.secondary')};
  background-color: ${prop('theme.navigation.toggle.primary')};
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: ${prop('theme.border.radius.fifty')};
  position: absolute;
  right: -10px;
  top: 122px;
  z-index: ${prop('theme.zindex.above')};

  &:hover {
    background-color: ${prop('theme.navigation.toggle.tertiary')};
  }
`

export const Navigation = ({
  height,
  toOpen,
  defaultStructure,
  defaultInterables,
  onInterable,
  onToggled
}) => {
  const [toggled, setToggled] = useState(false)
  const [hasNavigationHover, setHasNavigationHover] = useState(false)
  const [hasToggleHover, setHasToggleHover] = useState(false)
  const { animations, navigation } = useContext(ThemeContext)
  const containerRef = useRef(null)

  return (
    <>
      <Provider
        value={{
          defaultStructure,
          defaultInterables,
          onInterable
        }}
      >
        <Container
          data-testid="container"
          ref={containerRef}
          height={height}
          onMouseEnter={() => setHasNavigationHover(true)}
          onMouseLeave={() => setHasNavigationHover(false)}
          initialPose={animations.visible}
          pose={toggled ? animations.collapsed : animations.visible}
        >
          {!toggled && (
            <Switcher
              toOpen={toOpen}
              height={containerRef.current && containerRef.current.offsetHeight}
            />
          )}
          {hasNavigationHover && (
            <Toggleable
              data-testid="toggleable"
              onMouseEnter={() => setHasToggleHover(true)}
              onMouseLeave={() => setHasToggleHover(false)}
              onClick={() => {
                setToggled(!toggled)
                onToggled(!toggled)
              }}
            >
              <Icon
                name={toggled ? 'right' : 'left'}
                width={10}
                height={10}
                color={
                  hasToggleHover
                    ? navigation.toggle.secondary
                    : navigation.toggle.tertiary
                }
              />
            </Toggleable>
          )}
        </Container>
      </Provider>
    </>
  )
}

Navigation.Separator = Separator
Navigation.Type = Type
Navigation.Interable = Interable
Navigation.Title = Title
Navigation.Control = Control

Navigation.useInterable = useInterable

Navigation.defaultProps = {
  onInterable: () => {},
  onToggled: () => {},
  isReady: true
}

Navigation.propTypes = {
  toOpen: PropTypes.func,
  height: PropTypes.string,
  onInterable: PropTypes.func,
  onToggled: PropTypes.func,
  defaultInterables: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      interables: PropTypes.arrayOf(
        PropTypes.shape({
          id: PropTypes.string.isRequired,
          value: PropTypes.any.isRequired
        })
      )
    })
  ).isRequired,
  defaultStructure: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.func.isRequired,
      type: PropTypes.oneOf(['category']).isRequired,
      children: PropTypes.arrayOf(
        PropTypes.shape({
          parent: PropTypes.string.isRequired,
          id: PropTypes.string.isRequired,
          title: PropTypes.func.isRequired,
          type: PropTypes.oneOf(['group', 'item']).isRequired,
          prevent: PropTypes.func,
          onPrevent: PropTypes.func,
          isOpen: PropTypes.bool,
          children: PropTypes.arrayOf(
            PropTypes.shape({
              parent: PropTypes.string.isRequired,
              id: PropTypes.string.isRequired,
              title: PropTypes.func.isRequired,
              type: PropTypes.oneOf(['group', 'item']).isRequired,
              prevent: PropTypes.func,
              onPrevent: PropTypes.func,
              isOpen: PropTypes.bool,
              children: PropTypes.array
            })
          )
        })
      )
    })
  )
}
