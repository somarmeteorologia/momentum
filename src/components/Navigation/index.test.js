import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { Theme } from '@components/Theme'
import { Navigation } from '@components/Navigation'

import TesteableProvider from './TesteableProvider'

const NEWSLETTER_TITLE = 'Boletins'
const NEWSLETTER_CONTENT = 'Content from newsletter'

const TO_BACK = 'Voltar'

const CARS_CONTENT = 'Ford'

const withProvider = ({ toOpen }) => (
  <ThemeProvider theme={Theme.light}>
    <TesteableProvider>
      <Navigation toOpen={toOpen} />
    </TesteableProvider>
  </ThemeProvider>
)

test('Not expect visual regression', () => {
  const { container } = render(withProvider({}))

  expect(container.firstChild).toMatchSnapshot()
})

test('Expect to open on a specific group by default', () => {
  const toOpen = () => 'newsletter'

  const { queryByText } = render(withProvider({ toOpen }))

  expect(queryByText(NEWSLETTER_CONTENT)).toBeVisible()
  expect(queryByText(NEWSLETTER_TITLE)).toBeNull()
})

test('Expect back button must be working', () => {
  const toOpen = () => 'newsletter'

  const { queryByText } = render(withProvider({ toOpen }))

  expect(queryByText(NEWSLETTER_CONTENT)).toBeVisible()

  fireEvent.click(queryByText(TO_BACK))

  expect(queryByText(NEWSLETTER_CONTENT)).toBeNull()
  expect(queryByText(NEWSLETTER_TITLE)).toBeVisible()
})

test('Expect show and hide menu button must be working', () => {
  const { getByTestId, queryByText } = render(withProvider({}))

  expect(queryByText(NEWSLETTER_TITLE)).toBeVisible()

  fireEvent.mouseEnter(getByTestId('container'))
  fireEvent.click(getByTestId('toggleable'))
  expect(queryByText(NEWSLETTER_TITLE)).toBeNull()

  fireEvent.mouseEnter(getByTestId('container'))
  fireEvent.click(getByTestId('toggleable'))
  expect(queryByText(NEWSLETTER_TITLE)).toBeVisible()
})

test('Expect to navigate between groups', () => {
  const { getByTestId, queryByText } = render(withProvider({}))

  fireEvent.click(getByTestId('open-cars'))
  expect(queryByText(CARS_CONTENT)).toBeVisible()

  fireEvent.click(getByTestId('open-ford'))
  expect(queryByText('Unavailable')).toBeVisible()

  fireEvent.click(queryByText(TO_BACK))
  expect(queryByText(CARS_CONTENT)).toBeVisible()

  fireEvent.click(queryByText(TO_BACK))
  expect(queryByText(NEWSLETTER_TITLE)).toBeVisible()
})

test('Expect to interact with switchers', () => {})
