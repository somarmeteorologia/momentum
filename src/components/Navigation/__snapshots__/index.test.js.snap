// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not expect visual regression 1`] = `
.c2 {
  display: inline-block;
  margin-right: 10px;
}

.c4 {
  display: inline-block;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 0;
}

.c7 {
  display: inline-block;
}

.c15 {
  width: 100%;
  height: 1px;
  border: none;
  background-color: #d7dcf2;
  margin: 10px 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 15px;
  padding: 15px 10px;
  background-color: #d7dcf2;
}

.c6 {
  border-radius: 25px;
  height: 50px;
  line-height: 50px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 10px;
  padding: 10px 15px;
  margin: 0 auto;
  width: calc(100% - 20px);
}

.c6:hover {
  background-color: #fff;
}

.c6:hover path,
.c6:hover span {
  color: #133BEC;
}

.c14 {
  padding: 0 15px;
}

.c0 {
  background-color: #f1f3fA;
  height: 100%;
  position: relative;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: inline-block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  border-radius: 1.5rem;
  background-color: #133BEC;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-transition: background-color 0.5s ease-out;
  transition: background-color 0.5s ease-out;
  height: 15px;
  min-width: 24px;
  padding: 0 3px;
}

.c12 {
  border-radius: 50%;
  background-color: #fff;
  height: 10px;
  width: 10px;
}

.c13 {
  font-family: Inter,sans-serif;
  color: #080B19;
  cursor: pointer;
  font-size: 0.75rem;
  margin-left: 8px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c10 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c5 {
  font-family: Inter,sans-serif;
  font-weight: 600;
  display: block;
  color: #19244d;
  font-size: 0.875rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c17 {
  font-family: Inter,sans-serif;
  font-weight: 500;
  display: block;
  color: #080B19;
  font-size: 0.875rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  class="c0"
  data-testid="container"
  style="width: 270px;"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      height="20"
      width="20"
    >
      <div
        class="c3"
      />
    </div>
    <div
      class="c4"
      color="#19244d"
      display="block"
    >
      <span
        class="c5"
        color="#19244d"
        display="block"
      >
        Monitoramento
      </span>
    </div>
  </div>
  <div
    class="c6"
    data-testid="open-cars"
  >
    <div
      class="c7"
      data-interable="interable"
    >
      <div
        class="c8"
      >
        <label
          class="c9"
          for="cars"
        >
          <input
            checked=""
            class="c10"
            id="cars"
            type="checkbox"
          />
          <div
            class="c11"
          >
            <div
              class="c12"
            />
          </div>
          <span
            class="c13"
          >
            Cars
          </span>
        </label>
      </div>
    </div>
    <div
      class="c4"
      height="12"
      width="12"
    >
      <div
        class="c3"
      />
    </div>
  </div>
  <div
    class="c14"
  >
    <hr
      class="c15"
    />
  </div>
  <div
    class="c6"
    data-testid="open-newsletter"
  >
    <div
      class="c16"
    >
      <div
        class="c2"
        height="20"
        width="20"
      >
        <div
          class="c3"
        />
      </div>
      <div
        class="c4"
        display="block"
      >
        <span
          class="c17"
          display="block"
        >
          Boletins
        </span>
      </div>
    </div>
    <div
      class="c4"
      height="12"
      width="12"
    >
      <div
        class="c3"
      />
    </div>
  </div>
</div>
`;
